common/src/main/scala/com/gu/zuora/encoding/CapitalizationEncoder.scala

```diff
@@ -0,0 +1,34 @@
+package com.gu.zuora.encoding
+
+import cats.instances.vector._
+import com.gu.helpers.StringExtensions._
+import com.gu.i18n.Currency
+import io.circe.generic.decoding.DerivedDecoder
+import io.circe.generic.encoding.DerivedObjectEncoder
+import io.circe.generic.semiauto._
+import io.circe.{Decoder, Encoder, JsonObject, ObjectEncoder}
+import shapeless.Lazy
+
+object CapitalizationEncoder {
+  implicit val encodeCurrency: Encoder[Currency] = Encoder.encodeString.contramap[Currency](_.iso)
+
+  def decapitalizingDecoder[A](implicit decode: Lazy[DerivedDecoder[A]]): Decoder[A] =
+    deriveDecoder[A].prepare(
+      _.withFocus(
+        _.mapObject(decapitalizeFields)
+      )
+    )
+
+  def capitalizingEncoder[A](implicit encode: Lazy[DerivedObjectEncoder[A]]): ObjectEncoder[A] =
+    deriveEncoder[A].mapJsonObject(capitalizeFields)
+
+  def modifyFields(json: JsonObject)(f: String => String): JsonObject = {
+    val newFields = json.fields.map(str => f(str)).zip(json.values)
```

Comment:
Would this be more or less clear?

 `json.toVector.map { case (k, v) => f(k) -> v }`